packages/block-editor/src/hooks/block-style-variation.js

```diff
@@ -69,6 +70,9 @@ function OverrideStyles( { override } ) {
  * based on an incoming theme config. If a matching style is found in the config,
  * a new override is created and returned. The overrides can be used in conjunction with
  * useStyleOverride to apply the new styles to the editor.
+ * NOTE: This component is required to load global styles revisions config. Style variation overrides are
+ * generated using the canvas's current global styles (see useBlockStyleVariation()). This component is,
+ * however, due to be refactored and therefore it's use elsewhere is not recommended.
```

Comment:
Do you think this could be communicated through the component's name? 

I'm only wondering if `Experimental` gets ignored more than something like "unstable". It's hard to see another case where this would be reused before the the proposed updates to the global styles architecture though.

Feel free to ignore this comment.

Reply:
I'm open to that. I see `__unstable` used a lot for components. 

Something like `__unstableBlockStyleVariationOverridesWithConfig` might be a better warning.

Reply:
> It's hard to see another case where this would be reused before the the proposed updates to the global styles architecture though

I might have spoken too soon. It looks like the previews for patterns etc might also need some updates to reflect section styles.

Reply:
> It looks like the previews for patterns etc might also need some updates to reflect section styles.


Oh, good spotting! Let me know if I can help there too.

Reply:
Thanks, after a bit of digging it appears to be a slightly different issue to this PR. 

Here the variation style overrides were "lost" entirely after the client ids change and need to be regenerated. This other issue now appears to be a matter of global styles being passed to the preview overriding the style overrides for variations in the parent editor (outside the preview's iframe).

**TL;DR**
I don't think the other issue will be able to reuse much if any of the fix here so shouldn't hold this PR up at all.